## debug Tranception with retrieval

In [2]:
from os.path import exists
from os import chdir
chdir("/n/groups/marks/projects/marks_lab_and_oatml/ProteinGym2/")

directory = 'model_scores/zero_shot_substitutions/Tranception_L_retrieval/'
idx = -1
with open('EVCouplings/output/Beltran_Lehner_2025/temp_ref_file.csv') as f_in:
    for lines in f_in:
        dms_id = lines.split(',')[0]
        score_csv = directory + dms_id + '.csv'
        if not exists(score_csv):
            print(dms_id, idx)
        idx += 1

DMS_id -1


The length of MSA is smaller than MSA_end index

In [3]:
import pandas as pd
df = pd.read_csv('EVCouplings/output/Beltran_Lehner_2025/temp_ref_file.csv')
df

,DMS_id,MSA_bitscore,MSA_num_seqs,MSA_perc_cov,MSA_num_cov,MSA_N_eff,MSA_Neff_L,MSA_Neff_L_category,weight_file_name,MSA_filename,...,MSA_start,MSA_end,MSA_len,MSA_theta,taxon,source_organism,includes_multiple_mutants,DMS_wildtype_score,first_author,coarse_selection_type
0,A0A2R8Y422_HUMAN_Beltran_2025_PF00240_2,0.7,61559,0.971,68,5929.1,84.701140,medium,A0A2R8Y422_HUMAN_Beltran_2025_PF00240_2_theta_...,A0A2R8Y422_HUMAN_Beltran_2025_PF00240_2_theta_...,...,1,70,70,0.2,Human,Saccharomyces cerevisiae,False,0,Beltran,Abundance
1,FEZF1_HUMAN_Beltran_2025_PF00096_289,1.5,645870,0.955,21,1813.5,82.430408,medium,FEZF1_HUMAN_Beltran_2025_PF00096_289_theta_0.8...,FEZF1_HUMAN_Beltran_2025_PF00096_289_theta_0.8...,...,1,22,22,0.2,Human,Saccharomyces cerevisiae,False,0,Beltran,Abundance
2,SPD2B_HUMAN_Beltran_2025_PF00018_155,0.9,33216,0.945,52,721.9,13.125128,medium,SPD2B_HUMAN_Beltran_2025_PF00018_155_theta_0.8...,SPD2B_HUMAN_Beltran_2025_PF00018_155_theta_0.8...,...,1,55,55,0.2,Human,Saccharomyces cerevisiae,False,0,Beltran,Abundance
3,SPD2B_HUMAN_Beltran_2025_PF00018_222,0.9,45592,0.898,53,947.7,16.062838,medium,SPD2B_HUMAN_Beltran_2025_PF00018_222_theta_0.8...,SPD2B_HUMAN_Beltran_2025_PF00018_222_theta_0.8...,...,1,59,59,0.2,Human,Saccharomyces cerevisiae,False,0,Beltran,Abundance
4,A2RRE5_HUMAN_Beltran_2025_PF01846_267,0.7,4543,0.952,60,128.7,2.042930,medium,A2RRE5_HUMAN_Beltran_2025_PF01846_267_theta_0....,A2RRE5_HUMAN_Beltran_2025_PF01846_267_theta_0....,...,1,63,63,0.2,Human,Saccharomyces cerevisiae,False,0,Beltran,Abundance
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
514,CD2AP_HUMAN_Beltran_2025_PF14604_1,1.1,91191,0.881,52,1947.7,33.012110,medium,CD2AP_HUMAN_Beltran_2025_PF14604_1_theta_0.8_b...,CD2AP_HUMAN_Beltran_2025_PF14604_1_theta_0.8_b...,...,1,59,59,0.2,Human,Saccharomyces cerevisiae,False,0,Beltran,Abundance
515,CD2AP_HUMAN_Beltran_2025_PF14604_109,1.1,101933,0.869,53,1764.6,28.927503,medium,CD2AP_HUMAN_Beltran_2025_PF14604_109_theta_0.8...,CD2AP_HUMAN_Beltran_2025_PF14604_109_theta_0.8...,...,1,61,61,0.2,Human,Saccharomyces cerevisiae,False,0,Beltran,Abundance
516,NCOR2_HUMAN_Beltran_2025_PF00249_613,0.7,57518,0.828,48,1467.9,25.307762,medium,NCOR2_HUMAN_Beltran_2025_PF00249_613_theta_0.8...,NCOR2_HUMAN_Beltran_2025_PF00249_613_theta_0.8...,...,1,58,58,0.2,Human,Saccharomyces cerevisiae,False,0,Beltran,Abundance
517,USH1C_HUMAN_Beltran_2025_PF00595_207,0.7,123337,0.831,74,1477.5,16.601235,medium,USH1C_HUMAN_Beltran_2025_PF00595_207_theta_0.8...,USH1C_HUMAN_Beltran_2025_PF00595_207_theta_0.8...,...,1,89,89,0.2,Human,Saccharomyces cerevisiae,False,0,Beltran,Abundance


In [7]:
df[df['seq_len'] != df['MSA_end']][['DMS_id', 'MSA_end', 'seq_len']]

,DMS_id,MSA_end,seq_len
6,WWP2_HUMAN_Beltran_2025_PF00397_440,41,54
82,BAG3_HUMAN_Beltran_2025_PF00397_11,67,49
185,YAP1_HUMAN_Beltran_2025_PF00397_162,46,50
435,WWTR1_HUMAN_Beltran_2025_PF00397_116,56,47


Fix: Fix the bug when making the reference file and manually updated the MSA_end to be the same as seq_len

## Debug AIDO

CUX2_HUMAN_Beltran_2025_PF02376_1036

Issue: the a2m reader reads in sequences by line and cannot handle sequences that span more than 1 line

fixes are in `/n/groups/marks/projects/marks_lab_and_oatml/ProteinGym2/scoring_scripts/test/pg_baselines/AIDO/compute_fitness.py`

## Debug PoET

In [8]:
from pathlib import Path

In [14]:
ref_series = pd.read_csv("EVCouplings/output/Beltran_Lehner_2025/temp_ref_file.csv").iloc[0]
msa_filename = ref_series["MSA_filename"].removesuffix(".a2m")
msa_filepath = Path("DMS_assays/processed_data/Beltran_Lehner_2025/cleaned").joinpath(msa_filename + ".a3m.zst")

In [15]:
msa_filepath

PosixPath('DMS_assays/processed_data/Beltran_Lehner_2025/cleaned/A0A2R8Y422_HUMAN_Beltran_2025_PF00240_2_theta_0.8_b0.7.a3m.zst')

In [11]:
msa_filename

'A0A2R8Y422_HUMAN_Beltran_2025_PF00240_2_theta_0.8_b0.7'

## debug EVE

reference sheet: 28027 sequences

In [2]:
import numpy as np
from evcouplings.align import Alignment, map_matrix
from collections import defaultdict
import pandas as pd
import sys

In [5]:
msa_location = "EVCouplings/output/Beltran_Lehner_2025/selected_alignments/ODPX_HUMAN_Beltran_2025_PF02817_181_theta_0.8_b1.05.a2m"

In [4]:
with open("EVCouplings/output/Beltran_Lehner_2025/selected_alignments/RBP2_HUMAN_Beltran_2025_PF00641_1414_theta_0.8_b0.9.a2m", "r") as infile:
    aln = Alignment.from_file(infile, format="fasta")
    print(aln.N)

28027


In [5]:
weights_file = 'EVCouplings/output/Beltran_Lehner_2025/weights/RBP2_HUMAN_Beltran_2025_PF00641_1414_theta_0.8_b0.9.npy'
weights = np.load(weights_file)
print(weights.shape[0])

28004


Is it because of the 50% gap fraction filter?

In [6]:
focus_seq_name = ""
seq_name_to_sequence = defaultdict(str)

with open(msa_location, "r") as msa_data:
    for i, line in enumerate(msa_data):
        line = line.rstrip()
        if line.startswith(">"):
            name = line
            if i == 0:
                focus_seq_name = name
        else:
            seq_name_to_sequence[name] += line
print("Number of sequences in MSA (before preprocessing):", len(seq_name_to_sequence))

Number of sequences in MSA (before preprocessing): 28027


In [7]:
msa_df = pd.DataFrame.from_dict(seq_name_to_sequence, orient='index', columns=['sequence'])

In [11]:
msa_df

,sequence
>P49792_PF00641_1414/1-30,KKEGHWDCSICLVRNEPTVSRCIACQNTKS
>UniRef100_UPI0020B20602/1308-1335,KKEGEWECSVCCIRNSSTAMSCAACQNA--
>UniRef100_UPI0020B20602/1442-1469,KKPGQWDCGTCAVMNEATATSCIACQAP--
>UniRef100_UPI0020B20602/1500-1527,KKPGQWDCGTCAVMNEATATSCIACQAP--
>UniRef100_UPI0020B20602/1558-1585,KKPGQWDCGTCAVMNEATATSCIACQAP--
...,...
>UniRef100_A0A816F0L2/34-62,KNENQWQCLVCSMKNSASSRQCQTCDVPK-
>UniRef100_A0A817KEW8/416-443,KKENNWQCDECSGTNEPSTKRCVCCGTP--
>UniRef100_A0A7S3XIK0/106-132,--DGQWECSQCTLLNPSASASCGACRAAK-
>UniRef100_T0Q2R0/23-46,----MWECPTCTLRNTLTAEACDACGTT--


In [40]:
msa_df.sequence = msa_df.sequence.apply(lambda x: x.replace(".", "-")).apply(
    lambda x: ''.join([aa.upper() for aa in x]))
# Remove columns that would be gaps in the wild type
non_gap_wt_cols = [aa != '-' for aa in msa_df.sequence[focus_seq_name]]
msa_df['sequence'] = msa_df['sequence'].apply(
    lambda x: ''.join([aa for aa, non_gap_ind in zip(x, non_gap_wt_cols) if non_gap_ind]))

In [49]:
threshold_sequence_frac_gaps = 0.5

In [50]:
print("Calculating proportion of gaps")
msa_array = np.array([list(seq) for seq in msa_df.sequence])
gaps_array = np.array(list(map(lambda seq: [aa == '-' for aa in seq], msa_array)))
# Identify fragments with too many gaps
seq_gaps_frac = gaps_array.mean(axis=1)
seq_below_threshold = seq_gaps_frac <= threshold_sequence_frac_gaps
print("Proportion of sequences dropped due to fraction of gaps: " + str(
    round(float(1 - seq_below_threshold.sum() / seq_below_threshold.shape) * 100, 2)) + "%")

Calculating proportion of gaps
Proportion of sequences dropped due to fraction of gaps: 0.0%


/tmp/ipykernel_1016572/203036351.py:8: DeprecationWarning: Conversion of an array with ndim > 0 to a scalar is deprecated, and will error in future. Ensure you extract a single element from your array before performing this operation. (Deprecated NumPy 1.25.)
  round(float(1 - seq_below_threshold.sum() / seq_below_threshold.shape) * 100, 2)) + "%")


In [22]:
msa_array

array([['K', 'K', 'E', ..., 'T', 'K', 'S'],
       ['K', 'K', 'E', ..., 'A', '-', '-'],
       ['K', 'K', 'P', ..., 'P', '-', '-'],
       ...,
       ['-', '-', 'D', ..., 'A', 'K', '-'],
       ['-', '-', '-', ..., 'T', '-', '-'],
       ['-', '-', '-', ..., '-', '-', '-']],
      shape=(28027, 30), dtype='<U1')

In [23]:
gaps_array

array([[False, False, False, ..., False, False, False],
       [False, False, False, ..., False,  True,  True],
       [False, False, False, ..., False,  True,  True],
       ...,
       [ True,  True, False, ..., False, False,  True],
       [ True,  True,  True, ..., False,  True,  True],
       [ True,  True,  True, ...,  True,  True,  True]], shape=(28027, 30))

In [24]:
seq_gaps_frac

array([0.        , 0.06666667, 0.06666667, ..., 0.1       , 0.2       ,
       0.23333333], shape=(28027,))

all sequences have gap fraction < 50%

try filter by minimum column cov 70%

In [29]:
threshold_focus_cols_frac_gaps = 0.3

In [30]:
columns_gaps_frac = gaps_array[seq_below_threshold].mean(axis=0)
index_cols_below_threshold = columns_gaps_frac <= threshold_focus_cols_frac_gaps
print("Proportion of non-focus columns removed: " + str(
    round(float(1 - index_cols_below_threshold.sum() / index_cols_below_threshold.shape) * 100, 2)) + "%")

Proportion of non-focus columns removed: 13.33%


/tmp/ipykernel_1016572/430170202.py:4: DeprecationWarning: Conversion of an array with ndim > 0 to a scalar is deprecated, and will error in future. Ensure you extract a single element from your array before performing this operation. (Deprecated NumPy 1.25.)
  round(float(1 - index_cols_below_threshold.sum() / index_cols_below_threshold.shape) * 100, 2)) + "%")


In [31]:
index_cols_below_threshold

array([False, False,  True,  True,  True,  True,  True,  True,  True,
        True,  True,  True,  True,  True,  True,  True,  True,  True,
        True,  True,  True,  True,  True,  True,  True,  True,  True,
        True, False, False])

In [34]:
def _lower_case_and_filter_fragments(seq):
    return ''.join([aa.lower() if aa_ix in index_cols_below_threshold else aa for aa_ix, aa in enumerate(seq)])
msa_df['sequence'] = msa_df['sequence'].apply(
    lambda seq: ''.join([aa.upper() if upper_case_ind else aa.lower() for aa, upper_case_ind in
     zip(seq, index_cols_below_threshold)]))
msa_df = msa_df[seq_below_threshold]
# Overwrite seq_name_to_sequence with clean version
seq_name_to_sequence = defaultdict(str)
# Create a dictionary from msa_df.index to msa_df.sequence
seq_name_to_sequence = dict(zip(msa_df.index, msa_df.sequence))

In [37]:
num_sequences_removed_due_to_indeterminate_AAs = 0
num_sequences_before_indeterminate_AA_drop = len(seq_name_to_sequence)
alphabet_set = set(list("ACDEFGHIKLMNPQRSTVWY"))
seq_names_to_remove = []
for seq_name, sequence in seq_name_to_sequence.items():
    for letter in sequence:
        if letter not in alphabet_set and letter != "-":
            seq_names_to_remove.append(seq_name)
            continue
seq_names_to_remove = list(set(seq_names_to_remove))
for seq_name in seq_names_to_remove:
    num_sequences_removed_due_to_indeterminate_AAs+=1
    del seq_name_to_sequence[seq_name]
print("Proportion of sequences dropped due to indeterminate AAs: {}%".format(round(float(num_sequences_removed_due_to_indeterminate_AAs/num_sequences_before_indeterminate_AA_drop*100),2)))

Proportion of sequences dropped due to indeterminate AAs: 88.43%


In [39]:
len(seq_name_to_sequence)

3242

this is problematic, it shouldn't filter out this many sequences

In [3]:
# sys.path.append("/n/groups/marks/users/elain/PG2/ProteinGym")
from proteingym.utils.msa_utils import MSA_processing

In [11]:
def get_aln_stats(protein, MSA_location, weights_location, skip_one_hot_encodings=True):
    '''
    modified from get_aln_stats on https://github.com/aaronkollasch/EVE/blob/master/scripts/get_alignment_stats.py
    remove all AWS commands
    '''
    # weights_fname = protein + '_theta_' + str(theta) + '.npy'

    msa = MSA_processing(
        MSA_location=MSA_location,
        weights_location=weights_location,
        threshold_focus_cols_frac_gaps=1,
        use_weights=True
    )

    #assert msa.focus_seq_name.lstrip(">")  == protein
    num_seqs = 0
    with open(MSA_location) as f:
        for line in f:
            if line.startswith(">"):
                num_seqs += 1

    result = {
        "protein": protein,
        "num_seqs": msa.num_sequences,
        "num_seqs_unfiltered": num_seqs,
        "perc_filtered": msa.num_sequences / num_seqs,
        "n_eff": msa.Neff,
        "n_eff_l": msa.Neff / len(msa.focus_seq),
        "seq_len": len(msa.focus_seq),
        "num_cov": msa.one_hot_encoding.shape[1],
        "perc_cov": msa.one_hot_encoding.shape[1] / len(msa.focus_seq),
    }

    print(protein)
    return result

In [12]:
get_aln_stats("RBP2_HUMAN_Beltran_2025_PF00641_1414", "EVCouplings/output/Beltran_Lehner_2025/selected_alignments/RBP2_HUMAN_Beltran_2025_PF00641_1414_theta_0.8_b0.9.a2m", "./test.npy")

Proportion of sequences dropped due to fraction of gaps: 0.0%
Proportion of non-focus columns removed: 0.0%
Encoding sequences
One-hot encoded sequences shape:(28004, 30, 20)
Loaded sequence weights from disk
Neff = 1535.9666987077107
Data Shape = (28004, 30, 20)
RBP2_HUMAN_Beltran_2025_PF00641_1414


{'protein': 'RBP2_HUMAN_Beltran_2025_PF00641_1414',
 'num_seqs': 28004,
 'num_seqs_unfiltered': 28027,
 'perc_filtered': 0.9991793627573411,
 'n_eff': 1535.9666987077107,
 'n_eff_l': 51.19888995692369,
 'seq_len': 30,
 'num_cov': 30,
 'perc_cov': 1.0}

In [10]:
wt = np.load('EVCouplings/output/Beltran_Lehner_2025/weights/ODPX_HUMAN_Beltran_2025_PF02817_181_theta_0.8_b1.05.npy')

In [11]:
wt.shape[0]

1448

In [20]:
get_aln_stats("ODPX_HUMAN_Beltran_2025_PF02817_181", "EVCouplings/output/Beltran_Lehner_2025/selected_alignments/ODPX_HUMAN_Beltran_2025_PF02817_181_theta_0.8_b1.05.a2m", "./test2.npy")

Proportion of sequences dropped due to fraction of gaps: 0.0%
Proportion of non-focus columns removed: 0.0%
Encoding sequences
One-hot encoded sequences shape:(1447, 47, 20)
Computing sequence weights
Neff = 39.13101962965613
Data Shape = (1447, 47, 20)
ODPX_HUMAN_Beltran_2025_PF02817_181


{'protein': 'ODPX_HUMAN_Beltran_2025_PF02817_181',
 'num_seqs': 1447,
 'num_seqs_unfiltered': 1448,
 'perc_filtered': 0.9993093922651933,
 'n_eff': 39.13101962965613,
 'n_eff_l': 0.8325748857373645,
 'seq_len': 47,
 'num_cov': 47,
 'perc_cov': 1.0}

In [21]:
wt = np.load('./test2.npy')
wt.shape[0]

1447

issues with weight calculation on AWS. check which sequences have this issue

In [3]:
def check_weights_dim(protein, MSA_location, weights_location, issues_list, threshold_focus_cols_frac_gaps=1, skip_one_hot_encodings=True):
    msa = MSA_processing(
        MSA_location=MSA_location,
        weights_location=weights_location,
        threshold_focus_cols_frac_gaps=threshold_focus_cols_frac_gaps,
        use_weights=True
    )

    num_seqs = len(msa.seq_name_to_sequence)

    weights = np.load(weights_location)

    if num_seqs != weights.shape[0]:
        issues_list.append(f"{protein} has {num_seqs} processed sequences but {weights.shape[0]} entries in weights")

In [4]:
df = pd.read_csv("EVCouplings/output/Beltran_Lehner_2025/temp_ref_file.csv")
df.head()

,DMS_id,MSA_bitscore,MSA_num_seqs,MSA_perc_cov,MSA_num_cov,MSA_N_eff,MSA_Neff_L,MSA_Neff_L_category,weight_file_name,MSA_filename,...,MSA_start,MSA_end,MSA_len,MSA_theta,taxon,source_organism,includes_multiple_mutants,DMS_wildtype_score,first_author,coarse_selection_type
0,A0A2R8Y422_HUMAN_Beltran_2025_PF00240_2,0.7,61559,0.971,68,5929.1,84.701140,medium,A0A2R8Y422_HUMAN_Beltran_2025_PF00240_2_theta_...,A0A2R8Y422_HUMAN_Beltran_2025_PF00240_2_theta_...,...,1,70,70,0.2,Human,Saccharomyces cerevisiae,False,0,Beltran,Abundance
1,FEZF1_HUMAN_Beltran_2025_PF00096_289,1.5,645870,0.955,21,1813.5,82.430408,medium,FEZF1_HUMAN_Beltran_2025_PF00096_289_theta_0.8...,FEZF1_HUMAN_Beltran_2025_PF00096_289_theta_0.8...,...,1,22,22,0.2,Human,Saccharomyces cerevisiae,False,0,Beltran,Abundance
2,SPD2B_HUMAN_Beltran_2025_PF00018_155,0.9,33216,0.945,52,721.9,13.125128,medium,SPD2B_HUMAN_Beltran_2025_PF00018_155_theta_0.8...,SPD2B_HUMAN_Beltran_2025_PF00018_155_theta_0.8...,...,1,55,55,0.2,Human,Saccharomyces cerevisiae,False,0,Beltran,Abundance
3,SPD2B_HUMAN_Beltran_2025_PF00018_222,0.9,45592,0.898,53,947.7,16.062838,medium,SPD2B_HUMAN_Beltran_2025_PF00018_222_theta_0.8...,SPD2B_HUMAN_Beltran_2025_PF00018_222_theta_0.8...,...,1,59,59,0.2,Human,Saccharomyces cerevisiae,False,0,Beltran,Abundance
4,A2RRE5_HUMAN_Beltran_2025_PF01846_267,0.7,4543,0.952,60,128.7,2.042930,medium,A2RRE5_HUMAN_Beltran_2025_PF01846_267_theta_0....,A2RRE5_HUMAN_Beltran_2025_PF01846_267_theta_0....,...,1,63,63,0.2,Human,Saccharomyces cerevisiae,False,0,Beltran,Abundance


From the old PG repo (/n/groups/marks/projects/indels_human/anon/proteingym)

In [5]:
issues_list = []
row = df.iloc[6]
check_weights_dim(row['DMS_id'], "EVCouplings/output/Beltran_Lehner_2025/selected_alignments/"+row['MSA_filename'], 'EVCouplings/output/Beltran_Lehner_2025/weights/' + row['weight_file_name'], issues_list, 1)

NameError: name 'MSA_processing' is not defined

From the updated PG repo (/n/groups/marks/users/elain/PG2/ProteinGym/proteingym/)

In [6]:
import inspect
sys.path.insert(0, "/n/groups/marks/users/elain/PG2/ProteinGym/proteingym")
from utils.msa_utils import MSA_processing

print(f"Path to the module containing '{MSA_processing.__name__}': {inspect.getfile(MSA_processing)}")

issues_list = []
row = df.iloc[7]
check_weights_dim(row['DMS_id'], "EVCouplings/output/Beltran_Lehner_2025/selected_alignments/"+row['MSA_filename'], 'EVCouplings/output/Beltran_Lehner_2025/weights/' + row['weight_file_name'], issues_list, 1)

Path to the module containing 'MSA_processing': /n/groups/marks/users/elain/PG2/ProteinGym/proteingym/utils/msa_utils.py
Number of sequences in MSA (before preprocessing): 1448
Calculating proportion of gaps
Number of sequences dropped due to fraction of gaps: [0]
Number of non-focus columns removed: [0]
>UniRef100_A0A0D3L3P4/96-142
Number of sequences dropped due to indeterminate AAs: 1
Number of sequences after preprocessing: 1447
Data Shape = (1447, 47, 20)


In [7]:
check_weights_dim(row['DMS_id'], "EVCouplings/output/Beltran_Lehner_2025/selected_alignments/"+row['MSA_filename'], 'EVCouplings/output/Beltran_Lehner_2025/weights/' + row['weight_file_name'], issues_list, 0.3)

Number of sequences in MSA (before preprocessing): 1448
Calculating proportion of gaps
Number of sequences dropped due to fraction of gaps: [0]
Number of non-focus columns removed: [4]
Number of sequences dropped due to indeterminate AAs: 0
Number of sequences after preprocessing: 1448
Data Shape = (1448, 43, 20)


In [6]:
from proteingym.utils.msa_utils import MSA_processing
issues_list = []
for index, row in df.iterrows():
    check_weights_dim(row['DMS_id'], "EVCouplings/output/Beltran_Lehner_2025/selected_alignments/"+row['MSA_filename'], 'EVCouplings/output/Beltran_Lehner_2025/weights/' + row['weight_file_name'], issues_list)

Number of sequences dropped due to fraction of gaps: [0]
Number of non-focus columns removed: [0]


/n/groups/marks/projects/indels_human/anon/proteingym/utils/msa_utils.py:102: FutureWarning: Series.__getitem__ treating keys as positions is deprecated. In a future version, integer keys will always be treated as labels (consistent with DataFrame behavior). To access a value by position, use `ser.iloc[pos]`
  self.seq_name_to_sequence[msa_df.index[seq_idx]] = msa_df.sequence[seq_idx]


Number of sequences dropped due to indeterminate AAs: 420
Encoding sequences
One-hot encoded sequences shape:(61139, 70, 20)
Loaded sequence weights from disk
Neff = 5929.079783031528
Data Shape = (61139, 70, 20)
Number of sequences dropped due to fraction of gaps: [0]
Number of non-focus columns removed: [0]
Number of sequences dropped due to indeterminate AAs: 609
Encoding sequences
One-hot encoded sequences shape:(645261, 22, 20)
Loaded sequence weights from disk
Neff = 1813.46897642701
Data Shape = (645261, 22, 20)
Number of sequences dropped due to fraction of gaps: [0]
Number of non-focus columns removed: [0]
Number of sequences dropped due to indeterminate AAs: 31
Encoding sequences
One-hot encoded sequences shape:(33185, 55, 20)
Loaded sequence weights from disk
Neff = 721.8820344013982
Data Shape = (33185, 55, 20)
Number of sequences dropped due to fraction of gaps: [0]
Number of non-focus columns removed: [0]
Number of sequences dropped due to indeterminate AAs: 35
Encoding s

KeyboardInterrupt: 

In [ ]:
issues_list

['WWP2_HUMAN_Beltran_2025_PF00397_440 has 59924 processed sequences but 59926 entries in weights',
 'ODPX_HUMAN_Beltran_2025_PF02817_181 has 1447 processed sequences but 1448 entries in weights',
 'DVL1_HUMAN_Beltran_2025_PF00595_246 has 592050 processed sequences but 592066 entries in weights',
 'TCRG1_HUMAN_Beltran_2025_PF01846_645 has 11990 processed sequences but 11991 entries in weights',
 'NPHP1_HUMAN_Beltran_2025_PF00018_152 has 166704 processed sequences but 166708 entries in weights',
 'TRI18_HUMAN_Beltran_2025_PF00643_172 has 47718 processed sequences but 47719 entries in weights',
 'P73_HUMAN_Beltran_2025_PF07647_487 has 60301 processed sequences but 60304 entries in weights',
 'LDB3_HUMAN_Beltran_2025_PF00595_1 has 58044 processed sequences but 58045 entries in weights',
 'FLNB_HUMAN_Beltran_2025_PF00630_1423 has 175146 processed sequences but 175147 entries in weights',
 'FLNB_HUMAN_Beltran_2025_PF00630_2109 has 43342 processed sequences but 43344 entries in weights',
 'FL

In [8]:
protein_list = [a.split()[0] for a in issues_list]
print(protein_list)

['WWP2_HUMAN_Beltran_2025_PF00397_440', 'ODPX_HUMAN_Beltran_2025_PF02817_181', 'DVL1_HUMAN_Beltran_2025_PF00595_246', 'TCRG1_HUMAN_Beltran_2025_PF01846_645', 'NPHP1_HUMAN_Beltran_2025_PF00018_152', 'TRI18_HUMAN_Beltran_2025_PF00643_172', 'P73_HUMAN_Beltran_2025_PF07647_487', 'LDB3_HUMAN_Beltran_2025_PF00595_1', 'FLNB_HUMAN_Beltran_2025_PF00630_1423', 'FLNB_HUMAN_Beltran_2025_PF00630_2109', 'FLNB_HUMAN_Beltran_2025_PF00630_2289', 'CSDE1_HUMAN_Beltran_2025_PF00313_184', 'CSDE1_HUMAN_Beltran_2025_PF00313_23', 'MPDZ_HUMAN_Beltran_2025_PF00595_1860', 'MPDZ_HUMAN_Beltran_2025_PF00595_372', 'SASH1_HUMAN_Beltran_2025_PF00536_630', 'ABLM3_HUMAN_Beltran_2025_PF00412_149', 'ABLM3_HUMAN_Beltran_2025_PF00412_210', 'ZRAB2_HUMAN_Beltran_2025_PF00641_9', 'HERC2_HUMAN_Beltran_2025_PF11515_2555', 'STAU1_HUMAN_Beltran_2025_PF00035_183', 'ABL1_HUMAN_Beltran_2025_PF00018_64', 'SPTA1_HUMAN_Beltran_2025_PF00018_979', 'TNNC2_HUMAN_Beltran_2025_PF13499_1', 'VILI_CHICK_Beltran_2025_PF02209_762', 'LCK_HUMAN_Belt

In [9]:
len(protein_list)

145

In [15]:
wt = np.load('EVCouplings/output/Beltran_Lehner_2025/weights/ODPX_HUMAN_Beltran_2025_PF02817_181_theta_0.8_b1.05.npy')
wt.shape[0]

1448

In [14]:
",".join(str(c) for c in df.index[df['DMS_id'].isin(protein_list)])

'6,7,10,12,21,24,25,41,44,45,46,54,55,61,62,65,66,67,72,77,81,84,87,88,90,96,99,101,102,103,107,110,111,119,122,125,126,129,135,138,139,140,141,149,150,153,161,174,177,179,180,181,182,187,192,194,198,203,217,221,222,224,228,231,238,247,251,255,256,257,264,266,267,268,271,273,276,277,278,279,285,286,291,293,299,305,306,314,317,318,323,325,337,343,348,356,358,375,376,377,378,380,381,384,387,393,397,399,400,402,403,404,411,412,415,416,417,419,427,429,431,433,435,437,442,446,448,455,458,465,473,474,479,482,484,491,497,500,501,508,509,513,515,516,517'

Should rerun the Tranception with retrieval for all 145 protein sequences, because it also used the weights

TO-DO: can do a PR to the Tranception github repo, add an assertion line of the weights dimension and the processed MSA dimension